RefsViews/Branches/BranchesModel.hpp

```diff
@@ -50,9 +56,25 @@ class BranchesModel : public QAbstractItemModel
 signals:
     void gitError( const Git::Result& error );
 
+private slots:
+    void onRefCreated(RM::Repo* repo, RM::Ref* ref);
+    void onRefDestroyed(RM::Repo* repo, RM::Ref* ref);
+    void onRefMoved(RM::Repo* repo, RM::Ref* ref);
+
+private:
+    QModelIndex index(RefItem* item) const;
+
+    void insertRef(bool notify, const Git::Reference& ref);
+    inline RefItem* insertNamespace(const bool notify, RefItem* parent, const QString& name);
```

Comment:
Thanks a lot for the hint! These lines are subject to change anyways as I really need to  do the migration now.